packages/flutter_test/lib/src/mock_event_channel.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+/// A mock stream handler for an [EventChannel] that mimics the native
+/// StreamHandler API.
+abstract class MockStreamHandler {
+  /// Create a new [MockStreamHandler].
+  MockStreamHandler();
+
+  /// Create a new [InlineMockStreamHandler] with the given [onListen] and
+  /// [onCancel] handlers.
```

Comment:
Can we add a description of how to use the  `onListen`'s `MockStreamHandlerEventSink` in this doc comment and the one for `InlineMockStreamHandler` as well? Preferably with a code example.

It took me a couple reads of the changes to the tests to figure out how it was expected to be used, and a new user won't have the advantage of a diff of existing tests to help them understand how to use this.

Reply:
The `MockStreamHandlerEventSink` should have the exact same API as the native `EventSink` that it's mocking, so if users have written the native it should be pretty easy to understand. I'll still add comments for clarification though. If I'm missing something here and the API doesn't match the native `EventSink` this code should be rewritten so please let me know.

Reply:
Do you mean the Dart `EventSink` class? If so, the names for the methods on that class and `MockStreamHandlerEventSink` don't match up. If you are referring to a different `EventSink`, could you provide a link to documentation for that class in the docs as well?

Reply:
I'm referring to the EventSink that is used in the Android/iOS native code side of the EventChannel

Reply:
<img width="669" alt="Screenshot 2023-04-04 at 3 24 11 PM" src="https://user-images.githubusercontent.com/7896519/229898812-43597cdf-5ef0-4e91-8820-cc4754b0d029.png">

Reply:
Gotcha, thanks for clarifying. Could you add a link to the EventSink docs (https://api.flutter.dev/javadoc/io/flutter/plugin/common/EventChannel.EventSink.html or a similar link if you have a better reference) somewhere in the doc comment?

I don't think it's a safe assumption that the person writing or reading a test will be the same person that wrote the native side, so I'm looking for these docs to guide users to be able to write a test even if they have near zero experience with the native side.